### ISSUE TITLE
Fix Selected Date with DateInput Keyboard Commands

### ISSUE THREAD

**Issue by amandadupell** [grommet/grommet#5499]
### Expected Behavior

When the user navigates to the grid of dates using their keyboard commands, they should be able to use the arrow keys within the table to select a date properly and update the input field.

### Actual Behavior

When the user navigates to the grid of dates using their keyboard commands, they are able to use the arrow keys within the table, but when they try to select a day in the grid, it selects the "active" date and not the one the user is currently on. The input field is updated to the incorrect date.

### Video exhibiting the issue

https://user-images.githubusercontent.com/38971752/127914695-880a75ee-d955-48e5-afe0-e955aefe587c.mov

### Steps to Reproduce

1. Using VoiceOver on Mac or another screen reader, open the Form story for the DateInput component.
2. Navigate to the grid of dates using keyboard commands. 
3. Using the arrow key commands, navigate to a date in the grid.
4. Select a date and see how it is updated in the input field.

### Your Environment

- Grommet version: v2.17.4
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MacBook Pro

**Comment by taysea** [comment]
Interestingly, I found if I clicked ctrl+opt+space to select as voiceover prompted me to, then it worked correctly. However, if I just clicked "enter" then I was able to replicate the behavior shown here.

**Comment by MikeKingdom** [comment]
I know NVDA takes over the arrow keys unless your app has `role="application"`. Might be the same for this situation.

**Comment by taysea** [comment]
That is probably what we're running into. I tried adding `role="application"` to the Grommet wrapper on a DateInput storybook example I was trying, but when doing the same keyboard interaction I ran into the same problems... I wonder if given the fact that the selection does work with the key commands that voiceover is telling me, if we should consider it is working?

**Comment by taysea** [comment]
Dragging this to "Parking Lot" for now to wait for some feedback from Bill Tipton. As of now, it feels that DateInput is supporting the expected behavior. When using tab + arrow keys, selection can be made properly. When using ctrl+opt+arrow keys (VoiceOver "First-level" command) and selecting as is documented/read out by the screen reader, then selection is also made properly.

I am not sure that it would even be best practice to use ctrl+opt+arrow keys then attempt to press "enter" for selection. Amanda is going to do some follow up conversation with Bill to understand what he does, and we'll go from there. But likely we will be able to close this ticket.

**Comment by taysea** [comment]
Going to close this issue. Based on my research, DateInput is allowing the selection to function correctly. If user uses Voice-Over special commands, they should use the selection method from this mode. Bill Tipton uses a different screen reader than Voice Over, so that would be unrelated to this scenario.
